src/ApparentHorizons/StrahlkorperGr.cpp

```diff
@@ -220,17 +221,19 @@ Scalar<DataVector> spin_function(
     get(extrinsic_curvature_phi_normal) +=
         extrinsic_curvature_dot_normal * tangents.get(i, 1);
   }
-  get(sin_theta) = sin(ylm.theta_phi_points()[0]);
-  get(extrinsic_curvature_theta_normal_sin_theta) *= get(sin_theta);
-  get(extrinsic_curvature_phi_normal) *= get(sin_theta);
+
+  DataVector& sin_theta = temp;
+  sin_theta = sin(ylm.theta_phi_points()[0]);
+  get(extrinsic_curvature_theta_normal_sin_theta) *= sin_theta;
+  get(extrinsic_curvature_phi_normal) *= sin_theta;
 
   auto spin_function = make_with_value<Scalar<DataVector>>(sin_theta, 0.0);
 
```

Comment:
here say `Scalar<DataVector>& result = temp;` to avoid this allocation

then see the code @nilsdeppe provided to combine the rest into a single statement (that uses temp on both sides of the equation, but you can use the result reference on the left and the sin_theta ref on the right)

Reply:
Note that, because of the details of the return value optimization rules, you should put the actual variable name, rather than the name of a reference to it, in the return statement.

Reply:
Thanks again for the code suggestions, @nilsdeppe and @kidder. Can you please let me know if the current version needs further revision? If not, is this ready to squash and merge?